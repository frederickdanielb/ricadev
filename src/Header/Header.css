.custom-header-item{
    display: block;
    color: #222;
    font-size: 14px;
    font-weight: 900;
    line-height: 23px;
    text-decoration: none;
    text-transform: uppercase;
    padding: 2px 18px;
    letter-spacing: 0.07em;
    position: relative;
    border-radius: 4px;
}