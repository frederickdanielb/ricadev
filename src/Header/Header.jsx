import { Button, Col, Row } from "react-bootstrap";
import Menu from "./Menu";
import MenuItem from "./Menu/MenuItem";
import "bootstrap/dist/css/bootstrap.min.css";
import "./../index.css";
import "./Header.css";
const Header = ({
  logo,
  menuItems,
  loggedIn,
  loggedUserName,
  onClickLogo,
  onClickLoginButton,
  onClickLogoutButton,
}) => {
  const handleClickLogo = () => {
    if (onClickLogo) onClickLogo();
  };
  return (
    <header className="light_header">
      <div className="container">
        <Row>
          <Col>
            <div className="menu">
              <div className="brand-logo">
                <img
                  src={logo}
                  alt="brand-logo"
                  className="img-fluid lazyload"
                  onClick={handleClickLogo}
                />
              </div>
              {menuItems && (
                <Menu>
                  <>
                    {menuItems.map((item, index) => (
                      <MenuItem key={index} children={item.node} onClick={item.onClick} />
                    ))}
                  </>
                </Menu>
              )}
              <ul className="header-right">
                <li className={""}>
                  <span className={"custom-header-item"}>
                    {loggedIn && loggedUserName}
                  </span>
                </li>

                <li className="user">
                  {loggedIn && (
                    <Button variant="dark" onClick={onClickLogoutButton}>
                      Logout
                    </Button>
                  )}
                  {!loggedIn && (
                    <Button variant="dark" onClick={onClickLoginButton}>
                      Login
                    </Button>
                  )}
                </li>
              </ul>
            </div>
          </Col>
        </Row>
      </div>
    </header>
  );
};
export default Header;
