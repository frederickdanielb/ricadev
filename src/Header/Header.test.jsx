import {
	render,
	fireEvent,
	waitFor,
	screen,
	act,
} from "@testing-library/react";
import "@testing-library/jest-dom"
import { MockConfig } from "../../config/_mocks_/config";

import Header from "./Header";
import history from "../../history";
import { Router } from "react-router-dom";

describe("Header component", () => {
	const propsWithData = MockConfig;
	let container = null;

	beforeEach(() => {
		container = render(<Router history={history}><Header sessionUser={{name: "Test User"}} config={propsWithData} /></Router>);
	});

	it("should render correctly", async () => {
		expect(container.asFragment()).toMatchSnapshot();
	});
	it("should show params", () => {
		const username = screen.getByText(/Test User/)
		expect(username).toBeInTheDocument()
	})
	describe("and when component not have config", () => {
		beforeEach(() => {

			let propsWithoutData = {...MockConfig};
			container = render(<Router history={history}><Header sessionUser={null} config={propsWithoutData} /></Router>);
		});
		it("should not render params not mapped", () => {
			expect(container.asFragment()).toMatchSnapshot();
		});
	});
});
