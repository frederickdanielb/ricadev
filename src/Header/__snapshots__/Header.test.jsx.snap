// Vitest Snapshot v1

exports[`Header component > and when component not have config > should not render params not mapped 1`] = `
<DocumentFragment>
  <header
    class="light_header"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div
            class="menu"
          >
            <div
              class="brand-logo"
            >
              <a
                href="/hotels"
              >
                <img
                  alt=""
                  class="img-fluid lazyload"
                  src="https://www.panamericanaturismo.cl/_ns_img/logo/logoPanamericana200x70.png"
                />
              </a>
            </div>
            <nav>
              <div
                class="main-navbar"
              >
                <div
                  id="mainnav"
                >
                  <div
                    class="toggle-nav"
                  >
                    <i
                      class="fa fa-bars sidebar-bar"
                    />
                  </div>
                  <div
                    class="menu-overlay"
                  />
                  <ul
                    class="nav-menu"
                  >
                    <li
                      class="back-btn"
                    >
                      <div
                        class="mobile-back text-right"
                      >
                        <span>
                          Back
                        </span>
                        <i
                          aria-hidden="true"
                          class="fa fa-angle-right pl-2"
                        />
                      </div>
                    </li>
                    <li>
                      <a
                        aria-current="page"
                        class="nav-link menu-title active"
                        href="/"
                      >
                        Home
                      </a>
                    </li>
                    <li>
                      <a
                        class="nav-link menu-title"
                        href="/hotels"
                      >
                        Hotels
                      </a>
                    </li>
                  </ul>
                </div>
              </div>
            </nav>
            <ul
              class="header-right"
            >
              <li
                class="front-setting"
              >
                <select>
                  <option
                    value="Travelaway"
                  >
                    Travelaway
                  </option>
                  <option
                    value="Panamericana"
                  >
                    Panamericana
                  </option>
                  <option
                    value="Tiemponor"
                  >
                    Tiemponor
                  </option>
                  <option
                    value="Andariego"
                  >
                    Andariego
                  </option>
                </select>
              </li>
              <li
                class="user"
              >
                <a
                  class="btn btn-block btn-secondary"
                  href="/login"
                >
                  Login
                </a>
              </li>
              <li
                class="setting"
              >
                <a
                  href="/"
                >
                  <i
                    class="fas fa-cog"
                  />
                </a>
                <ul
                  class="setting-open"
                >
                  <li
                    class="front-setting"
                  >
                    <select>
                      <option
                        value="Travelaway"
                      >
                        Travelaway
                      </option>
                      <option
                        value="Panamericana"
                      >
                        Panamericana
                      </option>
                      <option
                        value="Tiemponor"
                      >
                        Tiemponor
                      </option>
                      <option
                        value="Andariego"
                      >
                        Andariego
                      </option>
                    </select>
                  </li>
                </ul>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;

exports[`Header component > should render correctly 1`] = `
<DocumentFragment>
  <header
    class="light_header"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div
            class="menu"
          >
            <div
              class="brand-logo"
            >
              <a
                href="/hotels"
              >
                <img
                  alt=""
                  class="img-fluid lazyload"
                  src="https://www.panamericanaturismo.cl/_ns_img/logo/logoPanamericana200x70.png"
                />
              </a>
            </div>
            <nav>
              <div
                class="main-navbar"
              >
                <div
                  id="mainnav"
                >
                  <div
                    class="toggle-nav"
                  >
                    <i
                      class="fa fa-bars sidebar-bar"
                    />
                  </div>
                  <div
                    class="menu-overlay"
                  />
                  <ul
                    class="nav-menu"
                  >
                    <li
                      class="back-btn"
                    >
                      <div
                        class="mobile-back text-right"
                      >
                        <span>
                          Back
                        </span>
                        <i
                          aria-hidden="true"
                          class="fa fa-angle-right pl-2"
                        />
                      </div>
                    </li>
                    <li>
                      <a
                        aria-current="page"
                        class="nav-link menu-title active"
                        href="/"
                      >
                        Home
                      </a>
                    </li>
                    <li>
                      <a
                        class="nav-link menu-title"
                        href="/hotels"
                      >
                        Hotels
                      </a>
                    </li>
                  </ul>
                </div>
              </div>
            </nav>
            <ul
              class="header-right"
            >
              <li
                class="front-setting"
              >
                <select>
                  <option
                    value="Travelaway"
                  >
                    Travelaway
                  </option>
                  <option
                    value="Panamericana"
                  >
                    Panamericana
                  </option>
                  <option
                    value="Tiemponor"
                  >
                    Tiemponor
                  </option>
                  <option
                    value="Andariego"
                  >
                    Andariego
                  </option>
                </select>
              </li>
              <li
                class=""
              >
                <span
                  class="custom-header-item"
                >
                  Test User
                </span>
              </li>
              <li
                class="user"
              >
                <a
                  class="btn btn-block btn-secondary"
                  href="/logout"
                >
                  Logout
                </a>
              </li>
              <li
                class="setting"
              >
                <a
                  href="/"
                >
                  <i
                    class="fas fa-cog"
                  />
                </a>
                <ul
                  class="setting-open"
                >
                  <li
                    class="front-setting"
                  >
                    <select>
                      <option
                        value="Travelaway"
                      >
                        Travelaway
                      </option>
                      <option
                        value="Panamericana"
                      >
                        Panamericana
                      </option>
                      <option
                        value="Tiemponor"
                      >
                        Tiemponor
                      </option>
                      <option
                        value="Andariego"
                      >
                        Andariego
                      </option>
                    </select>
                  </li>
                </ul>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
